XXX-module-import.md

```diff
@@ -0,0 +1,241 @@
+module.import('./dir').then(dir => {
+  dir.list('.').then(console.log);
+});
+```
+
+## Asynchronous exports: why?
+The example with core modules is mostly for
+copy and paste testability sake only,
+the truth is that many modules have
+asynchronous initialization behind databases connections,
+remote API calls, and other non blocking operations.
+
+New modules created with async `import(...)` in mind
+can early reject when needed, instead of emitting
+or throwing unhandled errors in the wild at distance.
+
+Importing a list of 20 modules at once, would be
+easy to _catch_ all at once and react accordingly,
+as opposite of handling every single module possible
+non blocking initialization a part.
+
+On top of that, on the client side,
+every module would most likely be preferred as asynchronous,
+so that this proposal can improve Universal-JS-ability even more.
+
+## Why not as global function ?
+The `import` word is reserved.
+Unless the community wants to wait until the official standard
+is finalized and the v8 change shipped,
+using a `global.import(path)` does not seem even
+aligned with the concept of module, which should never even use
+`global` reference if not to quickly feature test something,
+and does not seem to be universally friendly,
+since `global` is not standard outside NodeJS world.
+
+## Why polluting `module` and not `require` ?
+There are at least two reasons:
+
+  1. the Web has a role too
+  2. historical confusion and semantics
+
+The current proposal is aligned with what
+`<script type="module">` should also ship on browsers.
+This means that the word `module` is already part
+of both ECMAScript and Web specifications.
+
+The word `require`, and the presence of an extra callback
+that is strictly coupled with synchronous CommonJS only,
+something not considered by dynamic `import` at all
+and actually warned in console when similarly operated
+on the main thread of any Web page, would only create confusion
+in terms of what `module` and `import` means for the present
+and the future of the language, in a platform independent way.
+
+On top of that, `require` function has been historically
+[abused](https://www.npmjs.com/package/require.async) or
+[polluted](http://requirejs.org/docs/api.html#data-main)
+in an era where `Promise`, as well as `async` and `await`
+patterns, weren't even discussed in the JS community.
+
+Accordingly, adding yet another method to a function
+which name and semantics are unknown for the rest of
+the JS ecosystem, does not look like a good choice
+in terms of new, future friendly, module feature.
+
+On the other hand, being `module` where developers
+define their exports, I don't see any confusion
+in being `module` where developers can import
+their dependencies.
+
+After all, they are going to "_import module_"
+or "_export module_", and not going to
+"_require an import of a module_" since they never
+"_required an export of a module_".
+
+## Why as core feature ? Why not transpilers ?
```

Comment:
By the way, since you insist this should be core instead of being a userland package, I think this section should be amended to address that question.